lib/ddtrace/tracer.rb

```diff
@@ -160,22 +222,9 @@ def set_tags(tags)
     #   parent2 = tracer.trace('parent2')   # has no parent span
     #   parent2.finish()
     #
+    # This method accepts all the options accepted by \start_span.
     def trace(name, options = {})
-      span = Span.new(self, name, options)
-
-      # set up inheritance
-      parent = @buffer.get()
-      span.set_parent(parent)
-      @buffer.set(span)
-
-      @tags.each { |k, v| span.set_tag(k, v) } unless @tags.empty?
-
-      # sampling
-      if parent.nil?
-        @sampler.sample(span)
-      else
-        span.sampled = span.parent.sampled
-      end
+      span = start_span(name, options)
```

Comment:
At some point `trace` method has almost no logic left in, compared to the Python impl I moved https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/tracer.py#L220 `get_call_context` to `start_span`. I think it's better this way but I'm worried I missed something.